runtime/compilers/rillv1/parse_explore.go

```diff
@@ -165,18 +166,26 @@ func (p *Parser) parseExplore(node *Node) error {
 	// Build and validate time ranges
 	var timeRanges []*runtimev1.ExploreTimeRange
 	for _, tr := range tmp.TimeRanges {
-		if err := validateISO8601(tr.Range, false, false); err != nil {
+		if _, err := rilltime.Parse(tr.Range); err != nil {
 			return fmt.Errorf("invalid time range %q: %w", tr.Range, err)
 		}
 		res := &runtimev1.ExploreTimeRange{Range: tr.Range}
 		for _, ctr := range tr.ComparisonTimeRanges {
-			if err := validateISO8601(ctr.Offset, false, false); err != nil {
-				return fmt.Errorf("invalid comparison offset %q: %w", ctr.Offset, err)
-			}
+			isNewFormat := false
 			if ctr.Range != "" {
-				if err := validateISO8601(ctr.Range, false, false); err != nil {
+				rt, err := rilltime.Parse(ctr.Range)
+				if err != nil {
 					return fmt.Errorf("invalid comparison range %q: %w", ctr.Range, err)
 				}
+				isNewFormat = rt.IsNewFormat
+			}
+			if ctr.Offset != "" {
```

Comment:
Consider having a `rilltime.ParseCompatibility(ctr.Range, ctr.Offset)` function – this would avoid duplicating the error handling if `isoOffset` is combined with the new format, and also avoid exposing the `IsNewFormat` outside the package

Reply:
Update. Had to move `ValidateISO8601` to achieve this.